calico/common.py

```diff
@@ -435,34 +451,68 @@ def validate_endpoint(config, combined_id, endpoint):
         raise ValidationFailed(" ".join(issues))
 
 
-def validate_rules(profile_id, rules):
+def validate_tier_data(tier, data):
+    issues = []
+    if not VALID_ID_RE.match(tier):
+        issues.append("Invalid profile_id '%r'." % tier)
+
+    if not isinstance(data, dict):
+        raise ValidationFailed("Expected tier data to be a dict.")
+
+    if "order" in data:
+        order = data["order"]
+        if not isinstance(order, numbers.Number):
+            issues.append("Tier data contained non-numeric order field")
+    else:
+        issues.append("Missing 'order' field in tier data")
```

Comment:
You have to have an order but then we use lexicographic ordering as a tie-breaker.  So it's optional to use explicit ordering as a technique but the field isn't optional.